File: Eglantine/Data/Events/core_events.lua
--------------------
--core_events
--------------------
--This script holds common events to be called by the rest of the game,
--including picking up simple items and moving to doors

function pickup(item_name)
	runCoroutine(function()
		Event:MovePlayerTo(item_name)
		waitUntil("Player stopped")
		Event:GainItem(item_name)
		Event:DisableInteractable(item_name)
		Event:PlaySound("Extend")	
	end)
end

function door(door_name, target_room, target_entrance)
	runCoroutine(function()
		Event:MovePlayerTo(door_name)
		waitUntil("Player stopped")
		Event:ChangeRoom(target_room, target_entrance)
	end)
end

File: Eglantine/Scripts/RoomEvents/CoreEvents.cs
using System;
using System.Collections.Generic;
using ObjectivelyRadical.Scheduler;

namespace Eglantine.Engine
{
	public static partial class GameEvents
	{
		public static ScriptPauser waitSeconds(float time)
		{
			return ScriptPauser.WaitSeconds(time);
		}

		public static ScriptPauser waitUntil (string signal)
		{
			return ScriptPauser.WaitForSignal(signal);
		}

		public static IEnumerator<ScriptPauser> pickup(string itemName)
		{
			EventManager.Instance.MovePlayerTo(itemName);
			yield return waitUntil("Player stopped");

			EventManager.Instance.GainItem(itemName);
			EventManager.Instance.DisableInteractable(itemName);
			EventManager.Instance.PlaySound("Extend");
		}

		public static IEnumerator<ScriptPauser> door(string doorName, string targetRoom, string targetEntrance)
		{
			EventManager.Instance.MovePlayerTo(doorName);
			yield return waitUntil("Player stopped");
			EventManager.Instance.ChangeRoom(targetRoom, targetEntrance);
		}
	}
}

